### PR TITLE
Fix _geo point not being correctly parsed

### PR DESCRIPTION
**Pull request by bidoubiwa**
fixes: #143 

If a user provides a field named `_geo` with a `GeoPoint` type, the type will be transformed into a Meilisearch geopoint compatible format.

In firestore:
```
{
    "_geo": { "latitude": x, "longitude": y }
}
```

In Meilisearch it becomes:
```
{
    "_geo": { "lng": x, "lat": y }
}
```

### CODE REVIEW

**Comment by bidoubiwa** [comment]
remove in favor of #154

File: functions/src/meilisearch-adapter.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
+'use strict'
+/*
+ * Copyright 2022 Meilisearch
```
Comment (alallema) [inline functions/src/meilisearch-adapter.ts:3]: Do we have to change this every year?

File: functions/__tests__/functions.test.ts [hunk 2]
```diff
@@ -42,6 +42,7 @@ describe('extension', () => {
   beforeEach(() => {
     restoreEnv = mockedEnv(defaultEnvironment)
     config = require('../src/config').config
+    config.collectionPath = 'collection'
```
Comment (alallema) [inline functions/__tests__/functions.test.ts:45]: Is this new? What is it for?

File: functions/src/meilisearch-adapter.ts [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  return fieldsToIndex ? fieldsToIndex.split(/[ ,]+/) : []
+}
+
+/**
+ * Update special fields to Meilisearch compatible format
+ * @param {firestore.DocumentData} document
+ * @param {string[]} rawFieldsToIndex
+ * @return {firestore.DocumentData} A properly formatted array of field and value.
+ */
+export function adaptFields(
```
Comment (alallema) [inline functions/src/meilisearch-adapter.ts:74]: Wouldn't we rename it `adaptFieldsForMeilisearch` is the same time?

**Review by alallema — COMMENTED** [review]
Just a few questions, but Well Done!!!

File: functions/__tests__/functions.test.ts (hunk 2 above)
Comment (bidoubiwa) [reply functions/__tests__/functions.test.ts:45]: Not sure why it has been added. @oplik0 could maybe tell us more.

**Review by alallema — APPROVED** [review]
LGTM! 🎉

File: functions/__tests__/functions.test.ts (hunk 2 above)
Comment (oplik0) [reply functions/__tests__/functions.test.ts:45]: This is a result of a change in firebase functions event triggers (line 37) - `handler` was removed and instead one needs to use `document(path).onWrite`.

With the regex for the collection path (I think it only allowed odd number of path parts - so only paths to a collection and not a specific document) this should always result in a wildcard path for all documents in selected collection.

Tge tests here didn't have a collection path despite it technically being required by the extension, and while it worked before now the path is used in the listener, so it had to be set in the config for these tests.

**Comment by bidoubiwa** [comment]
bors merge

**Pull request merged**